### PR TITLE
Add configuration options for touchpads

### PR DESCRIPTION
**Pull request by BenJarg**
This may make more sense as a patch, but seems like a somewhat basic configuration option for touchpads (and has been asked for in #141).

### CODE REVIEW

**Comment by Sevz17** [comment]
It may worth adding more options, like changing the default accel speed, and also the default accel profile

**Comment by Sevz17** [comment]
It will definitely be by config.h

**Comment by BenJarg** [comment]
So now this configures disable while typing, the accel speed, and accel profile. Is there anything else reasonable that should be on this list?

**Comment by pm4rcin** [comment]
[Here's](https://wayland.freedesktop.org/libinput/doc/latest/configuration.html) all the available options. @Sevz17 do we want every option from there?

**Comment by BenJarg** [comment]
Thanks, perhaps left handed mode and middle button emulation should also be added. I suppose we should decide what exactly to include though, and whether or not we want everything available.

File: config.def.h [hunk 1]
```diff
@@ -50,6 +50,9 @@ static const int repeat_delay = 600;
 /* Trackpad */
 static const int tap_to_click = 1;
 static const int natural_scrolling = 0;
+static const int disable_while_typing = 1;
+static const enum libinput_config_accel_profile accel_profile = LIBINPUT_CONFIG_ACCEL_PROFILE_ADAPTIVE;
```
Comment (Sevz17) [inline config.def.h:54]: add a comment like:
```
You can choose between
LIBINPUT_CONFIG_ACCEL_PROFILE_FLAT
LIBINPUT_CONFIG_ACCEL_PROFILE_ADAPTIVE
```

**Comment by Sevz17** [comment]
- tap to click
- tap and drag
- drag lock
- left handled mode
- middle button emulation
- [libinput_device_config_scroll_set_method()](https://wayland.freedesktop.org/libinput/doc/latest/api/group__config.html#ga482951a2b1faf837e43d30d9c92dc9d3)
- [libinput_device_config_send_events_set_mode()](https://wayland.freedesktop.org/libinput/doc/latest/api/group__config.html#gaacc79635146b888ddaf1cb35363b22ce)
- [libinput_device_config_dwtp_set_enabled()](https://wayland.freedesktop.org/libinput/doc/latest/api/group__config.html#ga32a8b64126f3a03aabaab367a7dadc98)

I think these options should be configured, does anyone want to add or remove some of these?

**Comment by pm4rcin** [comment]
Basically we will have all the options for Touchpad so I think it's good. I think we should provide comments for every option so people don't have to look for docs etc.

**Comment by BenJarg** [comment]
Ok, I've added everything on that list (except dwtp), and comments with values for the options of things that you would need to look up. I didn't added dwtp because it looks like it won't be available until libinput 1.21, and the latest release is 1.20, so as far as I can tell it's a future setting (someone correct me if I misunderstood the docs).

File: dwl.c [hunk 2]
```diff
@@ -981,11 +981,50 @@ createpointer(struct wlr_input_device *device)
...
-		if (tap_to_click && libinput_device_config_tap_get_finger_count(libinput_device))
-			libinput_device_config_tap_set_enabled(libinput_device, LIBINPUT_CONFIG_TAP_ENABLED);
+		if (libinput_device_config_tap_get_finger_count(libinput_device)) {
+			if (tap_to_click)
+				libinput_device_config_tap_set_enabled(libinput_device, LIBINPUT_CONFIG_TAP_ENABLED);
+
+			if (tap_and_drag)
+				libinput_device_config_tap_set_drag_enabled(libinput_device, LIBINPUT_CONFIG_DRAG_ENABLED);
+			else
+				libinput_device_config_tap_set_drag_enabled(libinput_device, LIBINPUT_CONFIG_DRAG_DISABLED);
```
Comment (Sevz17) [inline dwl.c:991]: I think is better just `libinput_device_config_tap_set_drag_enabled(libinput_device, tap_and_drag);`

File: dwl.c [hunk 3]
```diff
@@ -981,11 +981,50 @@ createpointer(struct wlr_input_device *device)
...
+
+			if (drag_lock)
+				libinput_device_config_tap_set_enabled(libinput_device, LIBINPUT_CONFIG_TAP_ENABLED);
+			else
+				libinput_device_config_tap_set_enabled(libinput_device, LIBINPUT_CONFIG_TAP_DISABLED);
```
Comment (Sevz17) [inline dwl.c:996]: The same as above.

File: dwl.c [hunk 4]
```diff
@@ -981,11 +981,50 @@ createpointer(struct wlr_input_device *device)
...
+			libinput_device_config_dwt_set_enabled(libinput_device, disable_while_typing);
+
+		if (libinput_device_config_left_handed_is_available(libinput_device))
+			libinput_device_config_left_handed_set(libinput_device, left_handed);
+
+		if (libinput_device_config_middle_emulation_is_available(libinput_device)) {
+			if (middle_button_emulation)
+				libinput_device_config_middle_emulation_set_enabled(libinput_device, LIBINPUT_CONFIG_MIDDLE_EMULATION_ENABLED);
+			else
+				libinput_device_config_middle_emulation_set_enabled(libinput_device, LIBINPUT_CONFIG_MIDDLE_EMULATION_DISABLED);
```
Comment (Sevz17) [inline dwl.c:1012]: The same as above

**Comment by Sevz17** [comment]
Other thing, at least for me it's more understandable place the variables below the comment about it.

File: dwl.c [hunk 5]
```diff
@@ -981,11 +981,50 @@ createpointer(struct wlr_input_device *device)
...
+		}
+
+		if (libinput_device_config_left_handed_is_available(libinput_device))
+			libinput_device_config_left_handed_set(libinput_device, left_handed);
```
Comment (Sevz17) [inline dwl.c:1016]: This is called above.

File: dwl.c (hunk 2 above)
Comment (BenJarg) [reply dwl.c]: I wasn't sure if `LIBINPUT_CONFIG_DRAG_ENABLED` was guaranteed to be 1, etc. But if that's the case, you're right, this is much cleaner.  Perhaps the if statement for `tap_to_click` should be replaced with a similar line if `LIBINPUT_CONFIG_TAP_ENABLED` is always going to be 1?

**Comment by Sevz17** [comment]
> Perhaps the if statement for `tap_to_click` should be replaced with a similar line if `LIBINPUT_CONFIG_TAP_ENABLED` is always going to be 1?

Yes, do it.
Also, the comment at the end of the function should be replaced.

**Comment by Sevz17** [comment]
Thanks!

**Pull request merged**